sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlOptions.scala

```diff
@@ -66,9 +66,9 @@ private[sql] class XmlOptions(
   }
 
   val compressionCodec = parameters.get(COMPRESSION).map(CompressionCodecs.getCodecClassName)
-  val rowTagOpt = parameters.get(XmlOptions.ROW_TAG)
+  val rowTagOpt = parameters.get(XmlOptions.ROW_TAG).map(_.trim)
   require(!rowTagRequired || rowTagOpt.isDefined, s"'${XmlOptions.ROW_TAG}' option is required.")
```

Comment:
@sandip-db do you remember why we need `rowTagRequired` variable?

Reply:
I added it as part of my last [PR](https://github.com/apache/spark/pull/43416/files). This is because `rowTag` is required only for file read/write (`XmlFIleFormat`) and ignored by `from_xml`, `schema_of_xml` and `DataFrameReader.xml` that takes `Dataset` as argument.